pygmt/helpers/utils.py

```diff
@@ -115,34 +126,52 @@ def _validate_data_input(
     GMTInvalidInput
         If the data input is not valid.
     """
-    if data is None:  # data is None
-        if x is None and y is None:  # both x and y are None
-            if required_data:  # data is not optional
+    # Check if too much data is provided.
+    if data is not None and any(v is not None for v in (x, y, z)):
+        msg = "Too much data. Use either data or x/y/z."
+        raise GMTInvalidInput(msg)
+
+    # Determine the data kind if not provided.
+    kind = kind or data_kind(data, required=required_data)
+
+    # Determine the required number of columns based on the required_z flag.
+    required_cols = 3 if required_z else 1
```

Comment:
Most modules usually requires 2 columns but there are exceptions like `histogram` and `info` will works for 1 column. So the default value is 1 if `required_z` is False. 

Will refactor this part in PR #3369